### ISSUE TITLE
Title with Text/Suffix/Separator Doesn't Check for Functions

### ISSUE THREAD

**Issue by NewETown** [lookback/meteor-seo#2]
Found an issue when using title with an object:
```
     seo: {
           title: {
                text: function() {
                     return this.data().title;
                },
                separator: "|",
                suffix: function() {
                     return this.data().category;
                }
            }
```
This returns the error "Title has to be a string!"

However if you do this:
```
      seo: {
            title:  function() {
                return this.data().title + " | Some Text" <= 70 ? this.data().title + " | Some Text" : this.data().title;
            },
```
Everything works as expected.

**Comment by johanbrook** [comment]
Thanks for the find! If you pull down the code in the PR https://github.com/lookback/meteor-seo/pull/3, does that work out better for you?

**Comment by NewETown** [comment]
I'll check that in just a minute, another issue to report (let me know if you want me to open a new ticket):

Use this URL: http://beta.untuckedstyle.com/ambition/like%20bathing%2C%20motivation%20is%20recommended%20daily

In the Facebook debugger (https://developers.facebook.com/tools/debug/og/object/) and you'll see half of the right info. The title and description works but it doesn't seem to pull the picture in properly. The documentation says that you apply the image tag to both Facebook and Twitter but I'm not sure it's passing through. Likewise in the Twitter Card Validator (https://cards-dev.twitter.com/validator) the card won't render.

Probably good to include my SEO object too:
```
     seo: {
            title:  function() {
                return this.data().title + " | Untucked Style" <= 70 ? this.data().title + " | Untucked Style" : this.data().title;
            },
            description: function() {
                return this.data().short;
            },
            image: function() {
                return this.data().image_url;
            },
            url: function() {
                return BASE_URL+this.data().category+'/'+this.data().url;
            },
            type: 'article',
            meta: {
                type: 'article',
            },
            fb: {
                app_id: FB_APP_ID,
                admins: ["", ""]
            },
            og: {
                site_name: 'Untucked Style',
            },
            twitter: {
                card: "summary_large_image",
                site: "@UntuckedStyle",
                creator: "@UntuckedStyle",
            }
        }
```

**Comment by johanbrook** [comment]
When visiting the URL [Facebook sees](https://developers.facebook.com/tools/debug/og/echo?q=http%3A%2F%2Fbeta.untuckedstyle.com%2Fambition%2Flike%2520bathing%252C%2520motivation%2520is%2520recommended%2520daily), you'll see a `TypeError` echoed:

```
<p>TypeError: 'undefined' is not a function (evaluating 'addthis.layers.refresh()')
http://beta.untuckedstyle.com/a144ab03e20b6000e4308bf6673ea487d1c1ff5a.js:194
</p>
```

This makes the `meta` tags render in the `body` tag, and then Facebook can't parse them (thus the error message *Errors That Must Be Fixed*) :)

**Comment by NewETown** [comment]
Ah! Thank you! I'll pull down your PR tonight when I get back. I didn't even realize there was an error message in the Facebook output.

**Comment by NewETown** [comment]
@johanbrook 

Looks like I fixed that error but the tags are still rendering in the body, see here: https://developers.facebook.com/tools/debug/og/echo?q=http%3A%2F%2Fbeta.untuckedstyle.com%2Fambition%2Flike%2520bathing%252C%2520motivation%2520is%2520recommended%2520daily

Any ideas?

**Comment by johanbrook** [comment]
Are you sure there isn't something else affecting the rendering of the meta tags? Seems some `style` tags are also rendered within the `body` element – I don't know if that's intentional or not.

`meteor-seo` uses the [`blaze-meta`](https://github.com/yasinuslu/blaze-meta) package under the hood to set the actual tags in the `head`. You can see the relevant line of code here: https://github.com/yasinuslu/blaze-meta/blob/master/lib/meta.js#L119. Note that it's rendering the template into `document.head`.

Are there anything malicious logged in the JS console?

**Comment by johanbrook** [comment]
Viewing the source of the Spiderable link with `?_escaped_fragment_` in the end is making the meta tags showing up in the `head`. View the source of [beta.untuckedstyle.com/ambition/like%20bathing%2C%20motivation%20is%20recommended%20daily?_escaped_fragment_=](http://beta.untuckedstyle.com/ambition/like%20bathing%2C%20motivation%20is%20recommended%20daily?_escaped_fragment_=).

There is a lot of injected markup in the `head`. I'd also check the different exceptions in the console, as they might interfere.

I also noted you've got `<meta name="ROBOTS" content="NOINDEX, NOFOLLOW, NOARCHIVE">` in the code :)

**Comment by NewETown** [comment]
Rendering the `style` tags in the `body` is not intentional, I thought it did that by default. Is there a way I can trace back to find out what might be doing that? I don't have many packages installed so it shouldn't be hard to track down.

Is this package compatible with Spiderable? In all likely hood I probably don't have Spiderable configured properly, I'm very new to Meteor and am still trying to understand the ecosystem. I don't see any JS errors but I do see another instance of the addthis layers breaking which I thought I fixed (it doesn't show up in Facebook's debugger anymore). I'll go back and fix that tonight and see if anything else pops up.

I tell the robots to not follow me yet because this is the beta site, I don't want it indexed :). Thanks for all of your help!

**Comment by NewETown** [comment]
@johanbrook I made a few changes on my end but the meta tags still seem to be rendering in the body. Do you think it might be a problem with the fast-render package? Do you have any apps that use your SEO package with fast-render?

**Comment by johanbrook** [comment]
@NewETown 

> Is this package compatible with Spiderable?

From my tests, this package does fine with the latest Spiderable (2.0). I've updated the README to reflect that.

> Do you think it might be a problem with the fast-render package? Do you have any apps that use your SEO package with fast-render?

No clue, actually. I haven't tested this package with `fast-render`. As I've said, I use [`blaze-meta`](https://github.com/yasinuslu/blaze-meta) under the hood for rendering the actual meta tags, so any issues regarding the actual rendering should probably go into that repo.

I'm gonna close this issue for now, since the original issue topic has been solved.
